mempool/v1/mempool.go

```diff
@@ -215,7 +217,7 @@ func (txmp *TxMempool) CheckTx(tx types.Tx, cb func(*abci.Response), txInfo memp
 		txKey := tx.Key()
 
 		// Check for the transaction in the cache.
-		if !txmp.cache.Push(tx) {
+		if !txmp.cache.Push(tx.Key()) {
```

Comment:
```suggestion
		if !txmp.cache.Push(txKey) {
```
The key was already computed above